pkg/distro/fedora/images.go

```diff
@@ -675,6 +680,11 @@ func iotInstallerImage(workload workload.Workload,
 
 	img.Filename = t.Filename()
 
+	img.SquashfsCompression = "lz4"
+	if common.VersionGreaterThanOrEqual(img.OSVersion, VERSION_ROOTFS_SQUASHFS) {
```

Comment:
This is the kind of thing though that I think should be owned by the OS, not by this repo. Especially because a lot of the code to support this at runtime had to have landed in the OS and is versioned with it already.

Maybe something like just having a config file in the dracut-live package - or even having this repo inspect a systemd `.mount` unit as configured in that root for the Live OS, etc.

I'm mainly familiar with the FCOS LiveOS code, see e.g.
https://github.com/coreos/fedora-coreos-config/blob/e8073f8d1933fa22b79b7ccc43069264fe22a823/overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/live-generator#L73
And there probably what would be better is to move that to a static unit instead of being generated, and then code like this can parse it.

Reply:
Maybe? But such speculation is outside the scope of this PR. The way things are working now is this is defined in the distro specific code in images.